h5py/_conv.pyx

```diff
@@ -714,17 +715,26 @@ cdef int conv_vlen2ndarray(void* ipt,
     itemsize = H5Tget_size(outtype.id)
     if itemsize > H5Tget_size(intype.id):
         data = realloc(data, itemsize * size)
-    H5Tconvert(intype.id, outtype.id, size, data, NULL, H5P_DEFAULT)
+
+    if needs_bkg_buffer(intype.id, outtype.id):
+        back_buf = malloc(H5Tget_size(outtype.id)*size)
```

Comment:
Do we need to free this buffer again, like in the other function you modified below?

Also, let's use our `emalloc` wrapper, which is already used elsewhere in this module. This takes care of raising a Python MemoryError if we can't get memory.